frontend-ui/src/views/PipelineView.vue

```diff
@@ -167,8 +172,33 @@ async function loadData(limit: number, skip: number, filter?: string) {
 }
 
 async function loadLastRuns() {
-  // TODO: Load last runs
+  const schedule_names = Array.from(new Set(tasks.value.map((task) => task.schedule_name)))
+
+  // Set lastRunsLoaded to true immediately and avoid setting the loading state
+  // so UI can start displaying data as it arrives
+  // loadingStore.startLoading('Loading last runs...')
   lastRunsLoaded.value = true
+
+  // Process each schedule individually with a pause between requests
+  for (const schedule_name of schedule_names) {
+    if (schedule_name) {
+      const result = await scheduleStore.fetchSchedule(schedule_name)
+      if (result && result.most_recent_task) {
+        schedulesLastRuns.value[schedule_name] = result.most_recent_task
+      }
+      if (scheduleStore.errors.length > 0) {
+        for (const error of scheduleStore.errors) {
+          notificationStore.showError(`Failed to load last run for schedule ${schedule_name}: ${error}`)
+        }
+      }
+      // Add a 100ms pause between requests to avoid overwhelming the backend
+      await new Promise(resolve => setTimeout(resolve, constants.TASKS_LOAD_SCHEDULES_DELAY))
+
+    }
+  }
+
+
+  loadingStore.stopLoading()
```

Comment:
loadingStore was already not loading anymore unless I'm mistaken, I think we do not need this.